File: frontend/types/project.ts
export const PROJECT_STATUSES = [
    "Planned",
    "InProgress",
    "Completed",
] as const;

export type ProjectStatus = typeof PROJECT_STATUSES[number];

export const PROJECT_DIFFICULTIES = [
    "Beginner",
    "Intermediate",
    "Advanced",
] as const;

export type ProjectDifficulty = typeof PROJECT_DIFFICULTIES[number];

export const PROJECT_CATEGORIES = [
    "Robotics",
    "Electronics",
    "ComputerScience",
    "GameDevelopment",
    "WebDevelopment",
] as const;

export type ProjectCategory = typeof PROJECT_CATEGORIES[number];

export type Project = {
    id: number;
    title: string;
    shortDescription: string;
    content: string;
    status: ProjectStatus;
    technologies: string[];
    featuredOrder?: number;
    thumbnail: string;
    githubUrl: string;
    demoUrl: string;
    startDate?: Date;
    lastUpdated?: Date;
    category: ProjectCategory;
    difficulty: ProjectDifficulty;
    viewCount: number;
    slug: string;
};

export type CreateProjectRequest = {
    title: string;
    shortDescription: string;
    content: string;
    category: ProjectCategory;
    difficulty: ProjectDifficulty;
    status: ProjectStatus;
    githubUrl: string;
    demoUrl: string;
    thumbnail: string;
    technologies: string[];
};

export type UpdateProjectRequest = CreateProjectRequest;

File: backend/EngineeringLogs.Api/Models/Project.cs
using EngineeringLogs.Api.Models.Enums;

namespace EngineeringLogs.Api.Models;

public class Project
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string ShortDescription { get; set; } = string.Empty;

    public ProjectStatus Status { get; set; }

    public List<string> Technologies { get; set; } = new List<string>();

    public int? FeaturedOrder { get; set; }

    public string Thumbnail { get; set; } = string.Empty; // Url

    public string GithubUrl { get; set; } = string.Empty;

    public string DemoUrl { get; set; } = string.Empty;

    public DateTime? StartDate { get; set; }

    public DateTime? LastUpdated { get; set; }

    public ProjectCategory Category { get; set; }

    public ProjectDifficulty Difficulty { get; set; }

    public int ViewCount { get; set; }

    public string Slug { get; set; } = string.Empty;
}